pkg/noderesourcetopology/config.go

```diff
@@ -0,0 +1,124 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package noderesourcetopology
+
+import (
+	"k8s.io/klog/v2"
+
+	topologyv1alpha2 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha2"
+)
+
+const (
+	AttributeScope  = "topologyManagerScope"
+	AttributePolicy = "topologyManagerPolicy"
+)
+
+// TODO: can we reuse constants from k/k/pkg/kubelet/cm/topology_manager ?
+const (
+	ScopeContainer = "container"
+	ScopePod       = "pod"
+)
+
+// TODO: can we reuse constants from k/k/pkg/kubelet/cm/topology_manager ?
+const (
+	PolicyNone           = "none"
+	PolicyBestEffort     = "best-effort"
+	PolicyRestricted     = "restricted"
+	PolicySingleNUMANode = "single-numa-node"
+)
+
+func IsValidScope(scope string) bool {
+	if scope == ScopeContainer || scope == ScopePod {
+		return true
+	}
+	return false
+}
+
+func IsValidPolicy(policy string) bool {
+	if policy == PolicyNone || policy == PolicyBestEffort || policy == PolicyRestricted || policy == PolicySingleNUMANode {
+		return true
+	}
+	return false
+}
+
+type TopologyManagerConfig struct {
+	Scope  string
+	Policy string
+}
+
+func (conf TopologyManagerConfig) IsComplete() bool {
+	if conf.Scope == "" {
+		return false
+	}
+	if conf.Policy == "" {
+		return false
+	}
+	return true
+}
+
+func topologyManagerConfigFromNodeResourceTopology(nodeTopology *topologyv1alpha2.NodeResourceTopology) TopologyManagerConfig {
+	conf := TopologyManagerConfig{}
+	// backward compatibility (v1alpha2 and previous)
+	updateTopologyManagerConfigFromTopologyPolicies(&conf, nodeTopology.Name, nodeTopology.TopologyPolicies)
+	// preferred new configuration source (v1alpha2 and onwards)
+	updateTopologyManagerConfigFromAttributes(&conf, nodeTopology.Attributes)
+	return conf
+}
+
+func updateTopologyManagerConfigFromAttributes(conf *TopologyManagerConfig, attrs topologyv1alpha2.AttributeList) {
+	for _, attr := range attrs {
+		if attr.Name == AttributeScope && IsValidScope(attr.Value) {
+			conf.Scope = attr.Value
+			continue
+		}
+		if attr.Name == AttributePolicy && IsValidPolicy(attr.Value) {
+			conf.Policy = attr.Value
+			continue
+		}
```

Comment:
Can we add `TODO` to add `TopologyManagerPolicies` when `scheduler-plugins` supports k8s 1.26. Basically when this [PR](https://github.com/kubernetes-sigs/scheduler-plugins/pull/489) is merged

Reply:
Yeah, my bad :) comment updated

Reply:
sure thing, it was planned already (https://github.com/kubernetes-sigs/scheduler-plugins/pull/488/commits/b131bdada31d1be1547beab439a4b1b5476cad17#diff-3532db6795688cc2309165cf54c8ad8d256cb477ca3f9a5b0f778f85aabf1b49R132) but forgot to add TODOs here